src/main.tsx

```diff
@@ -68,13 +102,38 @@ const Main = () => {
       dispatch(resetNotification());
     }
   }, [notification, toast, dispatch]);
+
+  const MainComponent = () => {
+    switch (loginstatus) {
+      case LoginStatus.NotLogged:
+        return (
+          <Center h="97vh" flexDir="column">
+            <Center
+              position="fixed"
+              inset={0}
+              bg="rgba(0,0,0,0.9)"
+              zIndex="tooltip"
+            >
+              <Spinner size="xl" color="blue.400" />
+            </Center>
+          </Center>
+        );
+      case LoginStatus.Logged:
+        return <Component />;
+      case LoginStatus.Error:
+        return <Unauthorised />;
+      default:
+        return <></>;
+    }
+  };
+
   return (
     <>
       <Header />
       <Box minH="70vh">
         <SimpleSidebar />
         <Box ml={{ base: 0, md: 60 }} p="4">
-          <Component />
+          {MainComponent()}
```

Comment:
not all pages are "authorized-only". should be part of the Sitemap to decide which routes requires authentication and (next feature) permissions.

Reply:
this still force a redirect if user is not logged in. The redirect should be triggered ONLY if the Route is 'authorized only'

the <AuthenticatedOnly> component is responsible of triggering Login() if is not Logged. not this.

Reply:
RequireAuth => auth.Check() ? auth.Login() : <Childrens>

RequirePermission => <RequireAuth> if hasPermission then <Children> else <403>